modules/yarpc/yarpc.go

```diff
@@ -132,31 +98,31 @@ type dispatcherController struct {
 	stopError  error
 	startError error
 
-	configs    []*yarpcConfig
+	configs    []*configWrapper
 	handlers   []handlerWithDispatcher
 	dispatcher yarpc.Dispatcher
 }
 
 // Adds the config to the controller.
-func (c *dispatcherController) addConfig(config yarpcConfig) {
-	c.Lock()
-	defer c.Unlock()
+func (controller *dispatcherController) addConfig(config configWrapper) {
```

Comment:
This change is unrelated to this PR. Please make this change in a different PR, otherwise it adds too much noise to the PR.

Also, FWIW `c` or `ctl` would be considered more idiomatic to Go: https://github.com/golang/go/wiki/CodeReviewComments#receiver-names